client/my-sites/checklist/wpcom-checklist/checklist-banner.js

```diff
@@ -0,0 +1,155 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import Gridicon from 'gridicons';
+import PropTypes from 'prop-types';
+import React, { Children, Component } from 'react';
+import store from 'store';
+import { connect } from 'react-redux';
+import { find, get, reduce } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Card from 'components/card';
+import ChecklistShowShare from 'my-sites/checklist/share';
+import ChecklistBannerTask from './checklist-banner-task';
+import Gauge from 'components/gauge';
+import ProgressBar from 'components/progress-bar';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSiteOption, getSiteSlug } from 'state/sites/selectors';
+
+const storeKeyForNeverShow = 'sitesNeverShowChecklistBanner';
+
+export class ChecklistBanner extends Component {
+	static propTypes = {
+		translate: PropTypes.func.isRequired,
+		siteDesignType: PropTypes.string,
+		siteSlug: PropTypes.string,
+	};
+
+	state = { closed: false };
+
+	handleClose = () => {
+		const { siteId } = this.props;
+		const sitesNeverShowBanner = store.get( storeKeyForNeverShow ) || {};
+		sitesNeverShowBanner[ `${ siteId }` ] = true;
+		store.set( storeKeyForNeverShow, sitesNeverShowBanner );
```

Comment:
Ugh. I mean we're obviously trying to phase out `store`. This should really be using Redux to persist this kind of setting. (Definitely not for this PR, but maybe add `TODO` comment to the code?)